grpcbp/server_middlewares.go

```diff
@@ -0,0 +1,94 @@
+package grpcbp
+
+import (
+	"context"
+	"strings"
+
+	"github.com/reddit/baseplate.go/ecinterface"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/tracing"
+	"github.com/reddit/baseplate.go/transport"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+func InjectServerSpanInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (_ interface{}, err error) {
+		split := strings.Split(info.FullMethod, "/")
+		m := split[len(split)-1]
+		ctx, span := StartSpanFromGRPCContext(ctx, m)
+
+		if md, ok := metadata.FromIncomingContext(ctx); ok {
+			if values := md.Get(transport.HeaderTracingTrace); len(values) > 0 {
```

Comment:
you might consider using `== 1` instead of `> 0`

Reply:
I think we need to handle the case when there is more than one, i.e. reporting somehow that those headers are used incorrectly?